ext/random/csprng.c

```diff
@@ -61,14 +62,22 @@
 # include <sanitizer/msan_interface.h>
 #endif
 
-PHPAPI zend_result php_random_bytes(void *bytes, size_t size, bool should_throw)
+#define php_random_bytes_error(...) do { \
+		if (errstr) { \
+			snprintf(errstr, errstr_size, __VA_ARGS__); \
+		} \
+	} while (0)
```

Comment:
I was working towards removing macros from ext/random as much as possible, because they are intransparent and thus easily cause bugs. The latter is especially bad in the CSPRNG as a security sensitive piece of code.

Can we make `errstr` either `__attribute__((__nonnull__))` or do something like this:

```
char dummy[1];
if (!errstr) {
  errstr = dummy;
  errstr_size = sizeof(dummy);
}
```

and then unconditionally call `snprintf`?

Reply:
Makes sense :+1: